### PR TITLE
driver: keep options ordered

### PR DESCRIPTION
**Pull request by compnerd**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

The linker flags are order dependent.  Ensure that we preserve ordering.

### CODE REVIEW

**Comment by tkremenek** [comment]
@compnerd Is there a way for us to regression test this, somehow?

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by compnerd** [comment]
@tkremenek that's what I'm thinking about.  Id really prefer to have one.  Unless I've misunderstood some piece of code, the options are pushed in different ways based on the target.  This currently requires building for a non-MachO target.  Is it really preferable to have a test that is dependent on the build host?

**Comment by gribozavr** [comment]
If that would help, you can constrain the test to only run on Linux.

**Comment by compnerd** [comment]
Actually, I think I may have an idea of how to do something more convoluted to just test the behavior that we care about.  Ill try my hand at that, and hopefully I can get a test case for this.

**Comment by compnerd** [comment]
Okay, so I have a test case that should be independent of the target since it artificially constructs the entries, and ensures that resultant order is preserved.  (Prior to the change, the order would be inverted).  Thanks for the gentle nudge for the test case.

File: test/Driver/linker-order.ll [hunk 1]
```diff
@@ -0,0 +1,9 @@
+; RUN: llc -mtriple armv7--linux-gnueabihf -filetype obj -o - %s | swift-autolink-extract -o - - | FileCheck %s
```
Comment (tkremenek) [inline test/Driver/linker-order.ll]: Can we get a comment  in the test case that actually describes the purpose of this test?  Right now it is unclear by just looking at the test to discern what it is testing.

Comment (compnerd) [inline test/Driver/linker-order.ll]: Done.  Added a comment explaining the test.  I guess I never got into the habit of that working on LLVM.

**Comment by tkremenek** [comment]
Thank you!

**Comment by tkremenek** [comment]
@swift-ci test and merge

**Comment by compnerd** [comment]
Ugh, not sure how that slipped past.  Fixed the missing change for std::vector usage.

@tkremenek my pleasure :-).

**Comment by compnerd** [comment]
@tkremenek mind giving this another go?

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

File: test/Driver/linker-order.ll [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+; REQUIRES: autolink-extract
```
Comment (gribozavr) [inline test/Driver/linker-order.ll]: Should the test go into `test/AutolinkExtract/`?

File: test/Driver/linker-order.ll (hunk 1 above)
Comment (gribozavr) [inline test/Driver/linker-order.ll]: The RUN line should say `%target-swift-autolink-extract`

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by compnerd** [comment]
@tkremenek okay to merge?  Seems that @gribozavr terminated the merge job (which is why it reported the failure).

**Comment by gribozavr** [comment]
@compnerd I terminated the merge job because I added review comments.

> Should the test go into `test/AutolinkExtract/`?

> The RUN line should say `%target-swift-autolink-extract`.

File: test/Driver/linker-order.ll (hunk 2 above)
Comment (compnerd) [inline test/Driver/linker-order.ll]: Oh, I didn't even see that.  I like that idea very much.  Ill upload a change with that moved.

**Comment by compnerd** [comment]
@gribozavr oh, sorry, didnt see those earlier.  Thanks for pointing that out.  Addressed both comments.

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Comment by compnerd** [comment]
The failure seems unrelated to this change?

**Comment by tkremenek** [comment]
@swift-ci test and merge

**Comment by compnerd** [comment]
Hmm, the linux build failed with:

swift-build: error: exit(1): which swiftc

Im not sure why swift packaging tests would fail with that when that is used throughout the rest of the tests.

**Comment by gribozavr** [comment]
I think that's a problem with package tests.

**Pull request merged**